client/app/actions/account_action_creator.coffee

```diff
@@ -36,9 +36,17 @@ module.exports = AccountActionCreator =
                                          account.draftMailbox? and \
                                          account.trashMailbox?
 
+                modalTimer = setTimeout ->
+                    AppDispatcher.dispatch
+                        type: ActionTypes.CLOSE_MODAL,
+                        value: {id: account.id, mailboxID: account.inboxMailbox}
+                , 5000
+
+                # We pass the timer so we can store and clear it in RouterStore
                 AppDispatcher.dispatch
                     type: ActionTypes.ADD_ACCOUNT_SUCCESS
-                    value: {account, areMailboxesConfigured}
+                    value: {account, areMailboxesConfigured, modalTimer}
```

Comment:
Be careful of anti-pattern; Stores have nothings to do with actionCreator except listening to them. Deciding what action to do is the stuff of ... ActionCreators files :) (I know it seems obvious each time I explain it).

You should add a property to the store, such as `isWaitingForRedirections`, or maybe use `RequestsStore` that is used for this. 
Preventing action `close_modal` is not usefull; if you get the right data to know if modal is still opened or not when you listen to this event, you would not need adding `modalTimer` as an argument.